docs/asciidoc/modules/ROOT/pages/cypher-execution/cypher-based-procedures-functions.adoc

```diff
@@ -7,55 +7,23 @@
 
-Here is a simple example:
+NOTE: APOC provides also xref::overview/apoc.custom/apoc.custom.asFunction.adoc[apoc.custom.asFunction] and xref::overview/apoc.custom/apoc.custom.asProcedure.adoc[apoc.custom.asProcedure] procedures, 
+but they have been deprecated in favor of `apoc.custom.declare*`s.
 
-[source,cypher]
-----
-CALL apoc.custom.asProcedure('answer','RETURN 42 as answer')
-----
-
-This registers the statement as procedure `custom.answer` that you then can call.
-As no information on parameter and return types is given, it just returns a stream of columns of maps called `row`.
-
-[source,cypher]
-----
-CALL custom.answer YIELD row
-RETURN row.answer
-----
-
-The same is possible as a function:
-
-[source,cypher]
-----
-CALL apoc.custom.asFunction('answer','RETURN 42')
-----
 
-NOTE: If you override procedures or functions you might need to call `call dbms.clearQueryCaches()` as lookups to internal id's are kept in compiled query plans.
+The first parameter of the `apoc.custom.declareProcedure` and `apoc.custom.declareFunction` procedures,
+is the signature of the procedure/function you want to create.
+This looks similar to the `signature` results returned by the `SHOW PROCEDURES YIELD signature` and `SHOW FUNCTIONS YIELD signature` cypher commands,
+that is:
+- for a procedure: `nameProcedure(firstParam = defaultValue :: typeParam , secondParam = defaultValue :: typeParam, ....) :: (firstResult :: typeResult, secondResult :: typeResult, ... )`
+- for a function: `nameFunction(firstParam = defaultValue :: typeParam , secondParam = defaultValue :: typeParam, ....) :: typeResult`
 
-== Custom Procedures with `apoc.custom.asProcedure`
+Note that, for both procedures and functions, the `= defaultValue` are optionals.
 
-Given statement will be registered as a procedure, the results will be turned into a stream of records.
-
-.Parameters
-[%autowidth,opts=header,cols="m,m,a"]
-|===
-| name
-| default
-| description
-
-| name | none | dot-separated name, will be prefixed with `custom`
-| statement | none | cypher statement to run, can use $parameters
-| mode | read | execution mode of the procedure: read, write, or schema
-| outputs | [["row","MAP"]] | List of pairs of name-type to be used as output columns, need to be in-order with the cypher statement, the default is a special case, that will collect all columns of the statement result into a map
-| inputs | [["params","MAP","{}"]] | Pairs or triples of name-type-default, to be used as input parameters. The default just takes an optional map, otherwise they will become proper paramters in order
-| description | "" | A general description about the business rules implemented into the procedure
-|===
-
-The type names are what you would expect and see in outputs of `dbms.procedures` or `apoc.help` just without the `?`.
-The default values are parsed as JSON.
+The `type`s are what you would expect to see in outputs of `SHOW PROCEDURES`, `SHOW FUNCTIONS` or `apoc.help` just without the `?`.
```

Comment:
I think it would b clearer to move this sentence down under the '.Type Names' heading and reformulate it to something like :

> The `typeParam` and `typeResult` in the signature parameter are what you expect to ... without the `?`. The following values are supported: